### PR TITLE
EOS-18797: [cortx-utils] Add confstore key validator in confstore

### PR DESCRIPTION
**Pull request by pratyush-seagate**
commit 7a1a1ed2f94a4165650255f491eb5d1f5bed28f3
Author: pratyush-seagate <pratyush.k.khan@seagate.com>
Date:   Tue Mar 16 15:01:33 2021 -0600

        EOS-18797: [cortx-utils] Add confstore key validator in confstore
            Change description:
                A confstore key validator v_confkeys, associated unit tests, update readme and usage info fix in commands.py
            Added/modified/deleted files:
                modified:   py-utils/README.md
                modified:   py-utils/src/utils/validator/commands.py
                new file:   py-utils/src/utils/validator/v_confkeys.py
                modified:   py-utils/src/utils/validator/validate.py
                new file:   py-utils/test/validator/test_confkeys_validator.py
            Unit test:
                    Run new associated added unit tests.
                        python3.6 py-utils/test/validator/test_confkeys_validator.py
                        ..
                        ----------------------------------------------------------------------
                        Ran 2 tests in 0.007s

                        OK

    Signed-off-by: pratyush-seagate <pratyush.k.khan@seagate.com>

-----
[View rendered py-utils/README.md](https://github.com/Seagate/cortx-utils/blob/br/pk/EOS-18798-confstore-key-validator/py-utils/README.md)

### CODE REVIEW

File: py-utils/src/utils/validator/commands.py [hunk 1]
```diff
@@ -219,3 +219,20 @@ def process(self):
         """Validate Path, file and directory check related status."""
 
         self._path.validate(self.v_path, self.args)
+
+class ConfKeysVCommand(VCommand):
+    """Verify if constore contains the provided keys."""
```
Comment (amitwac2608) [inline py-utils/src/utils/validator/commands.py:224]: typo in 'constore'

File: py-utils/src/utils/validator/commands.py [hunk 2]
```diff
@@ -219,3 +219,20 @@ def process(self):
...
+
+    def __init__(self, args):
+        super(ConfKeysVCommand, self).__init__(args)
+
+        from v_confkeys import ConfKeysV
+
+        self._confkeys = ConfKeysV()
+
+    def process(self):
+        """Validate if constore contains the provided keys."""
```
Comment (amitwac2608) [inline py-utils/src/utils/validator/commands.py:236]: typo

**Review by amitwac2608 — COMMENTED** [review]
minor comments, look good.

File: py-utils/src/utils/validator/v_confkeys.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (sachinpunadikar) [inline py-utils/src/utils/validator/v_confkeys.py:4]: correct the year 
2020 -> 2021

File: py-utils/test/validator/test_confkeys_validator.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (sachinpunadikar) [inline py-utils/test/validator/test_confkeys_validator.py:4]: correct the year

**Review by sachinpunadikar — APPROVED** [review]
Minor comments, please work on that.
Approving

**Comment by pratyush-seagate** [comment]
> 
> 
> Minor comments, please work on that.
> Approving

thanks. Please check now.

File: py-utils/src/utils/validator/commands.py [hunk 5]
```diff
@@ -219,3 +219,20 @@ def process(self):
...
+        super(ConfKeysVCommand, self).__init__(args)
+
+        from v_confkeys import ConfKeysV
+
+        self._confkeys = ConfKeysV()
+
+    def process(self):
+        """Validate if confstore contains the provided keys."""
+
+        self._confkeys.validate(self.v_confkeys, self.args)
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/commands.py:238]: Where is self.v_confkeys initialized ? shouldnt it be self.v_type ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/commands.py (hunk 5 above)
Comment (pratyush-seagate) [reply py-utils/src/utils/validator/commands.py]: thanks for catching this, fixed.

**Review by ujjwalpl — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Pull request merged**